### ISSUE TITLE
iperf3: error - control socket has closed unexpectedly

### ISSUE THREAD

**Issue by TJFOE** [esnet/iperf#1803]
Using iperf (v2) I get a correct answer about a slow connection in a 1GBit/s network from Manjaro to Xigmanas (NFS).

iperf 2.2.0-1
iperf 3.17.1

`System:
  Kernel: 6.6.54-2-MANJARO arch: x86_64 bits
  Desktop: GNOME v: 46.5 tk: GTK v: 3.24.43 wm: gnome-shell`

Complete System attached:
[System_complete.txt](https://github.com/user-attachments/files/17933522/System_complete.txt)

I tried several settings, but could not go out of the iperf3 error:

`[tom@tom-manjaro ~]$ iperf3 -c 192.168.0.43 -p 2049
iperf3: error - control socket has closed unexpectedly`

`[tom@tom-manjaro ~]$ iperf3 -c 192.168.0.43 -p 2049 -V
iperf 3.17.1
Linux tom-manjaro 6.6.54-2-MANJARO #1 SMP PREEMPT_DYNAMIC Tue Oct  8 03:11:08 UTC 2024 x86_64
Control connection MSS 1448
iperf3: error - control socket has closed unexpectedly`

`[tom@tom-manjaro ~]$ iperf -c 192.168.0.43 -p 2049
Client connecting to 192.168.0.43, TCP port 2049
TCP window size: 16.0 KByte (default)
[  1] local 192.168.0.2 port 48904 connected with 192.168.0.43 port 2049
[ ID] Interval       Transfer     Bandwidth
[  1] 0.0000-16.7145 sec  22.5 MBytes  11.3 Mbits/sec`

Any idea?

**Comment by TJFOE** [comment]
No, I don't use a iperf3-server? I have a problem with the speed from the Linux-PC to the NAS (Nfs4). So I was looking for iperf to measure the speed. The iperf2 command is the last one above in my first post.

I have this server-output:

Nov 27 08:04:36	xigmanas64fu	lighttpd: (h1.c.413) oversized request-header -> sending Status 431
Nov 27 08:07:10	xigmanas64fu	kernel: Limiting closed port RST response from 242 to 209 packets/sec
Nov 27 08:07:11	xigmanas64fu	kernel: Limiting closed port RST response from 277 to 203 packets/sec
Nov 27 08:07:12	xigmanas64fu	nfsd: accept failed: Software caused connection abort
Nov 27 08:07:13	xigmanas64fu	kernel: Limiting closed port RST response from 290 to 197 packets/sec
Nov 27 08:07:15	xigmanas64fu	kernel: Limiting closed port RST response from 289 to 185 packets/sec
Nov 27 08:07:16	xigmanas64fu	kernel: Limiting closed port RST response from 217 to 195 packets/sec
Nov 27 08:09:02	xigmanas64fu	lighttpd: (h1.c.413) oversized request-header -> sending Status 431
Nov 27 08:44:26	xigmanas64fu	root: AUTH: admin logged in from IP address 192.168.0.2
Nov 27 10:21:07	xigmanas64fu	root: AUTH: admin logged in from IP address 192.168.0.2
Nov 27 10:51:36	xigmanas64fu	lighttpd: (h1.c.413) oversized request-header -> sending Status 431
Nov 27 10:52:14	xigmanas64fu	lighttpd: (h1.c.413) oversized request-header -> sending Status 431
Nov 27 10:56:54	xigmanas64fu	lighttpd: (h1.c.413) oversized request-header -> sending Status 431
Nov 27 11:23:15	xigmanas64fu	root: AUTH: admin logged in from IP address 192.168.0.2

**Comment by TJFOE** [comment]
These ports are open at 192.168.0.43 (NAS):

Port	Status	Dienst
22/tcp	open	ssh
80/tcp	open	http
111/tcp	open	rpcbind
139/tcp	open	netbios-ssn
445/tcp	open	microsoft-ds
2049/tcp	open	nfs

**Comment by davidBar-On** [comment]
As I wrote, you cannot use ipef3 client with iperf (iperf2) server. Since you are using iperf server, and as you wrote the iperf test is successful, I am not sure what is the problem.

I the problem is that you want to use iperf3 instead, you must run iperf3 server.

**Comment by TJFOE** [comment]
I don't understand you. I have only both iperf-versions installed on my PC from which I try to measure the conncection to the NAS. I don't know anything about iperf-server. I think I am missing something to know...? :-)

**Comment by davidBar-On** [comment]
iperf (all versions) are testing the connection throughput, etc. using client-server interface.  In your case, iperf server is running on 192.168.0.43 machine and is using port 2049.  Since 192.168.x.x is a local address, I assume you can find the machine and what is running on it.

To understand this better, try the following in your PC:
1. In one window run iperf3 server by: `iperf3 -s`
2. In a second window run iperf3 client that is connecting to the server you run: `iperf3 -c localhost`

You should see the test results output from both the client and the server.

Note that the default port used is 5201.  If there is a problem with this port, you can try using another port using the `-p` option (for both the server and the client).

**Comment by TJFOE** [comment]
Awesome! Thank you! First I did it with success on the PC with 2 terminals open.

I found that on den XigmaNAS iperf3 is still installed! I didn't know this. Now I did this:

In the server terminal: iperf3 -s
PC: iperf3 -c 192.168.0.43

I got an answer and a middle Bitrate about 50 Mbits/sec. Do I right now? This Bitrate is still low. But it is not on iperf3...

**Comment by davidBar-On** [comment]
Looks good!

**Comment by TJFOE** [comment]
Thank you very much!

**Comment by rmelotte** [comment]
For anyone with the right permissions: it looks like this issue can be closed?
